sdk/nodejs/s3/bucket.ts

```diff
@@ -474,7 +475,7 @@ export interface BucketState {
     /**
      * A website object (documented below).
      */
-    readonly website?: pulumi.Input<{ errorDocument?: pulumi.Input<string>, indexDocument?: pulumi.Input<string>, redirectAllRequestsTo?: pulumi.Input<string>, routingRules?: pulumi.Input<string> }>;
+    readonly website?: pulumi.Input<{ errorDocument?: pulumi.Input<string>, indexDocument?: pulumi.Input<string>, redirectAllRequestsTo?: pulumi.Input<string>, routingRules?: pulumi.Input<string | RoutingRules> }>;
```

Comment:
Ideally, I'd like this to be expressed as:

```typescript
routingRules?: pulumi.Input<string | RoutingRule[]>
```

Rather than:

```typescript
routingRules?: pulumi.Input<string | RoutingRules>

// with the following from routingRules.ts:
export type RoutingRules = RoutingRule[];
```

But I had to create the `RoutingRules` placeholder type because I couldn't express `"RoutingRule[]"` as an `AltTypes` in the tfbridge.SchemaInfo`.

When trying to do so:

```go
AltTypes:  []tokens.Type{awsType(s3Mod+"/routingRules", "RoutingRule[]")},
```

It tries to import the array (which isn't valid TypeScript):

```typescript
import {RoutingRule[]} from "./routingRules";
```

(It should just import `RoutingRule`).

I could make a targeted change in `pulumi-terraform`'s `generate_nodejs.go` to strip the `[]` suffix when importing the type.

https://github.com/pulumi/pulumi-terraform/blob/2bd38b436a69b4056ae58ce8accc0dd5ff63a980/pkg/tfgen/generate_nodejs.go#L959

Changing this:

```go
imports[relmod][string(ct.Name())] = true
```

To this:

```go
imports[relmod][strings.TrimSuffix(string(ct.Name()), "[]")] = true
```

Trying out that change appears to work perfectly, but I'm slightly hesitant to do so as `ct.Name()` is supposed to conform to the `[A-Za-z_.][A-Za-z0-9_.]*` regex, even though the `[]` suffix appears to be getting through when I set it in the `AltTypes`.

Reply:
Honestly, `RoutingRules` seems fine to me. 

But if we can fix this in `pulumi-Terraform` to allow `[]` that’s great too.

Reply:
I'll just fix `pulumi-terraform` to all `[]` so we don't have to add unnecessary types that just alias the array. I'll add a commit to https://github.com/pulumi/pulumi-terraform/pull/393